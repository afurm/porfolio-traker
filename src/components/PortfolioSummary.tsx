import React from 'react';
import { Box, Typography, Container } from '@mui/material';

export interface PortfolioSummaryProps {
  totalWorth: number;
  totalProfitLoss: number;
  profitLossPercentage: number;
}

const PortfolioSummary: React.FC<PortfolioSummaryProps> = ({ totalWorth, totalProfitLoss, profitLossPercentage }) => {
  return (
    <Container maxWidth="sm" sx={{ p: 4, backgroundColor: 'background.paper', borderRadius: 2, boxShadow: 3 }}>
      <Typography variant="h4" align="center" gutterBottom>
        Portfolio Summary
      </Typography>
      <Box display="flex" flexDirection="column" gap={3}>
        <Box display="flex" justifyContent="space-between">
          <Typography>Total Worth:</Typography>
          <Typography fontWeight="bold">${totalWorth.toFixed(2)}</Typography>
        </Box>
        <Box display="flex" justifyContent="space-between">
          <Typography>Total Profit/Loss:</Typography>
          <Typography fontWeight="bold" color={totalProfitLoss >= 0 ? 'green' : 'red'}>
            ${totalProfitLoss.toFixed(2)} ({profitLossPercentage.toFixed(2)}%)
          </Typography>
        </Box>
      </Box>
    </Container>
  );
};

export default PortfolioSummary;