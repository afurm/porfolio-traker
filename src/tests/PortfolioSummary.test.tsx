import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import PortfolioSummary, { PortfolioSummaryProps } from '@/components/PortfolioSummary';

describe('PortfolioSummary Component Snapshot Test', () => {
  it('renders correctly with positive profit', () => {
    const mockProps: PortfolioSummaryProps = {
      totalWorth: 10000,
      totalProfitLoss: 2000,
      profitLossPercentage: 20,
    };

    const { asFragment } = render(<PortfolioSummary {...mockProps} />);

    // Take a snapshot of the initial rendering
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with negative profit', () => {
    const mockProps: PortfolioSummaryProps = {
      totalWorth: 10000,
      totalProfitLoss: -1500,
      profitLossPercentage: -15,
    };

    const { asFragment } = render(<PortfolioSummary {...mockProps} />);

    // Take a snapshot of the rendering when profit/loss is negative
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with zero profit', () => {
    const mockProps: PortfolioSummaryProps = {
      totalWorth: 10000,
      totalProfitLoss: 0,
      profitLossPercentage: 0,
    };

    const { asFragment } = render(<PortfolioSummary {...mockProps} />);

    // Take a snapshot of the rendering when profit/loss is zero
    expect(asFragment()).toMatchSnapshot();
  });
});
