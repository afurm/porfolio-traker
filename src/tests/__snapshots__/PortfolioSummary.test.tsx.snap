// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PortfolioSummary Component Snapshot Test renders correctly with negative profit 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthSm css-75k7kp-MuiContainer-root"
  >
    <h4
      class="MuiTypography-root MuiTypography-h4 MuiTypography-alignCenter MuiTypography-gutterBottom css-1ke64am-MuiTypography-root"
      style="--Typography-textAlign: center;"
    >
      Portfolio Summary
    </h4>
    <div
      class="MuiBox-root css-1i43dhb"
    >
      <div
        class="MuiBox-root css-gg4vpm"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1edfpdg-MuiTypography-root"
        >
          Total Worth:
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1ub6n32-MuiTypography-root"
        >
          $10000.00
        </p>
      </div>
      <div
        class="MuiBox-root css-gg4vpm"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1edfpdg-MuiTypography-root"
        >
          Total Profit/Loss:
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-12e3s4q-MuiTypography-root"
        >
          $-1500.00 (-15.00%)
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`PortfolioSummary Component Snapshot Test renders correctly with positive profit 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthSm css-75k7kp-MuiContainer-root"
  >
    <h4
      class="MuiTypography-root MuiTypography-h4 MuiTypography-alignCenter MuiTypography-gutterBottom css-1ke64am-MuiTypography-root"
      style="--Typography-textAlign: center;"
    >
      Portfolio Summary
    </h4>
    <div
      class="MuiBox-root css-1i43dhb"
    >
      <div
        class="MuiBox-root css-gg4vpm"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1edfpdg-MuiTypography-root"
        >
          Total Worth:
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1ub6n32-MuiTypography-root"
        >
          $10000.00
        </p>
      </div>
      <div
        class="MuiBox-root css-gg4vpm"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1edfpdg-MuiTypography-root"
        >
          Total Profit/Loss:
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-10tsa1k-MuiTypography-root"
        >
          $2000.00 (20.00%)
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`PortfolioSummary Component Snapshot Test renders correctly with zero profit 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthSm css-75k7kp-MuiContainer-root"
  >
    <h4
      class="MuiTypography-root MuiTypography-h4 MuiTypography-alignCenter MuiTypography-gutterBottom css-1ke64am-MuiTypography-root"
      style="--Typography-textAlign: center;"
    >
      Portfolio Summary
    </h4>
    <div
      class="MuiBox-root css-1i43dhb"
    >
      <div
        class="MuiBox-root css-gg4vpm"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1edfpdg-MuiTypography-root"
        >
          Total Worth:
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1ub6n32-MuiTypography-root"
        >
          $10000.00
        </p>
      </div>
      <div
        class="MuiBox-root css-gg4vpm"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1edfpdg-MuiTypography-root"
        >
          Total Profit/Loss:
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-10tsa1k-MuiTypography-root"
        >
          $0.00 (0.00%)
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
